config/backend-config/types.go

```diff
@@ -0,0 +1,167 @@
+	ID                    string
+	Name                  string
+	DestinationDefinition DestinationDefinitionT
+	Config                map[string]interface{}
+	Enabled               bool
+	Transformations       []TransformationT
+	IsProcessorEnabled    bool
+}
+
+type SourceT struct {
+	ID                         string
+	Name                       string
+	SourceDefinition           SourceDefinitionT
+	Config                     map[string]interface{}
+	Enabled                    bool
+	WorkspaceID                string
+	Destinations               []DestinationT
+	WriteKey                   string
+	DgSourceTrackingPlanConfig DgSourceTrackingPlanConfigT
+	Transient                  bool
+}
+
+type WorkspaceRegulationT struct {
+	ID             string
+	RegulationType string
+	WorkspaceID    string
+	UserID         string
+}
+
+type SourceRegulationT struct {
+	ID             string
+	RegulationType string
+	WorkspaceID    string
+	SourceID       string
+	UserID         string
+}
+
+type ConfigT struct {
+	EnableMetrics   bool            `json:"enableMetrics"`
+	WorkspaceID     string          `json:"workspaceId"`
+	Sources         []SourceT       `json:"sources"`
+	Libraries       LibrariesT      `json:"libraries"`
+	ConnectionFlags ConnectionFlags `json:"flags"`
+}
+
+type ConnectionFlags struct {
+	URL      string          `json:"url"`
+	Services map[string]bool `json:"services"`
+}
+
+type WRegulationsT struct {
+	WorkspaceRegulations []WorkspaceRegulationT `json:"workspaceRegulations"`
+	Start                int                    `json:"start"`
+	Limit                int                    `json:"limit"`
+	Size                 int                    `json:"size"`
+	End                  bool                   `json:"end"`
+	Next                 int                    `json:"next"`
+}
+
+type SRegulationsT struct {
+	SourceRegulations []SourceRegulationT `json:"sourceRegulations"`
+	Start             int                 `json:"start"`
+	Limit             int                 `json:"limit"`
+	Size              int                 `json:"size"`
+	End               bool                `json:"end"`
+	Next              int                 `json:"next"`
+}
+
+type TransformationT struct {
+	VersionID string
+	ID        string
+	Config    map[string]interface{}
+}
+
+type LibraryT struct {
+	VersionID string
+}
+
+type LibrariesT []LibraryT
+
+type DgSourceTrackingPlanConfigT struct {
+	SourceId            string                            `json:"sourceId"`
+	SourceConfigVersion int                               `json:"version"`
+	Config              map[string]map[string]interface{} `json:"config"`
+	MergedConfig        map[string]interface{}            `json:"mergedConfig"`
+	Deleted             bool                              `json:"deleted"`
+	TrackingPlan        TrackingPlanT                     `json:"trackingPlan"`
+}
+
+func (dgSourceTPConfigT *DgSourceTrackingPlanConfigT) GetMergedConfig(eventType string) map[string]interface{} {
```

Comment:
Should we be having these functions in the `types.go` file?

Reply:
The rationale behind this code "relocation" is that the `backend-config.go` was getting way too big, so I decided to move all (or almost all?) types along with their methods in `types.go`.

`GetMergedConfig` and `fetchEventConfig` don't depend much on anything else, they seemed pretty self explanatory and just tied to the `DgSourceTrackingPlanConfigT` type so I put them there as well. If you want them back then I'll have to also move the `DgSourceTrackingPlanConfigT` type along with them :thinking: